storage: better error when stores are throttled

tbg opened the pull request:
This is an annoying blocker during replication, and has also played a
role in test flakes such as #35307.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/35308)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35308#-:-L_9RdwO78Lr3SS1fT8p:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/storage/store_snapshot.go, line 755 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/35308#-L_9BQOpDsUidLVhpQiV:-L_9BQOpDsUidLVhpQiW:bwp3gjc) ([raw file](https://github.com/cockroachdb/cockroach/blob/44a9481c08d281b2d7de0b7fe9059787bc7e2250/pkg/storage/store_snapshot.go#L755)):*
> ```Go
> 			}
> 			err := &benignError{errors.Errorf("%s: remote declined %s: %s", to, snap, declinedMsg)}
> 			storePool.throttle(throttleDeclined, declinedMsg, to.StoreID)
> ```

Seems like you can generate the err after the call to `storePool.throttle` in which case you can `return &benignError{...}`. On the other hand, perhaps you mean to pass in `err.Error()` here as you do below. 


<!-- Sent from Reviewable.io -->

tbg committed:
storage: better error when stores are throttled

This is an annoying blocker during replication, and has also played a
role in test flakes such as #35307.

Release note: None

tbg commented:
Done, TFTR!

bors r=petermattis
